metrics.go

```diff
@@ -11,27 +11,35 @@ import (
 // with values relative to that last second.
 type MetricsBag map[string]interface{}
 
-// AddRawCounter adds a counter that always increments.
+// AddRawCounter adds a counter. A counter is a cumulative metrics that can only increase.
+// value must be positive.
 func (bag MetricsBag) AddRawCounter(name string, value int64) {
 	bag["c:"+name] = value
 }
 
-// AddDeltaCounter adds a count of something that happened in the last second
+// AddDeltaCounter adds a new delta increment for a counter. The increment represents
+// the variation of the counter in the current time interval.
+// delta must be positive.
 func (bag MetricsBag) AddDeltaCounter(name string, delta int64) {
 	bag["d:"+name] = delta
 }
 
-// AddGauge takes a snapshot of a value.
+// AddGauge adds a new snapshot of a value. A Gauge represents a single numerical
+// data point that can arbitrarily go up and down.
 func (bag MetricsBag) AddGauge(name string, value float64) {
 	bag["g:"+name] = value
 }
 
 // AddHistogram adds a set of values to track their statistical distribution.
+// A histogram samples observations and counts them in different 'buckets' in order
```

Comment:
```suggestion
// A histogram sample observes and counts them in different 'buckets' in order
```
Not sure I fully understand what you mean and thus if my suggestion is correct

Reply:
`sample` is the verb and `observation` the noun. Thus, I think that was correct.